experimental/packages/web-common/src/SessionManager.ts

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SessionIdGenerator } from './types/SessionIdGenerator';
+import { Session } from './types/Session';
+import { SessionProvider } from './types/SessionProvider';
+import { SessionObserver } from './types/SessionObserver';
+import { SessionStore } from './types/SessionStore';
+import { SessionPublisher } from './types/SessionPublisher';
+
+export interface SessionManagerConfig {
+  /** Class responsible for generating a session ID */
+  sessionIdGenerator: SessionIdGenerator;
+  /** Class responsible for storing session data over multiple page loads.  */
+  sessionStore: SessionStore;
+  /** Maximum duration of a session in seconds */
+  maxDuration?: number;
+  /** Maximum time without a user activity after which a session should be expired, in seconds */
+  inactivityTimeout?: number;
+}
+
+/**
+ * SessionManager is responsible for managing the active session including starting,
+ * resetting, and persisting.
+ */
+export class SessionManager implements SessionProvider, SessionPublisher {
```

Comment:
Hmm, is there any reason for users to inherit from `SessionManager`? 🤔 
If yes, would they be able to accomplish the same goal by using composition instead? 

If everything can be done can be done with composition, I'd recommend only exporting a factory function that returns a `SessionProvider & SessionPublisher` - this way we can cut down on the public API quite a bit which will simplify future changes.